physical/raft/raft.go

```diff
@@ -1089,6 +1085,11 @@ type SetupOpts struct {
 	// RecoveryModeConfig is the configuration for the raft cluster in recovery
 	// mode.
 	RecoveryModeConfig *raft.Configuration
+
+	// EffectiveSDKVersion is typically the version string baked into the binary.
+	// We pass it in though because it can be overridden in tests or via ENV in
+	// core.
+	EffectiveSDKVersion string
```

Comment:
I preferred making this an option when we setup the cluster since that is an existing mechanism for runtime config that isn't known at `NewRaftBackend` time rather than a customer `Set*` method that needs correct plumbing.